### PR TITLE
More consistent rp integration logging and rp configuration story

### PR DESCRIPTION
**Pull request by trask**
* Unifies app svc "user-logging" with regular "self diagnostics"
* Rename "attach" to "rp integration"
* Removes a couple of env vars that we aren't using
* Sets default self diagnostic file path for Azure Functions on Linux to `"/var/log/applicationinsights/" + DEFAULT_NAME`
* Do not read `applicationinsights.json` from agent directory for rp integrations because that makes it difficult to merge applicationinsights.json from RP and user json from `APPLICATIONINSIGHTS_CONFIGURATION_CONTENT`

### CODE REVIEW

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/wasbootstrap/configuration/Configuration.java [hunk 1]
```diff
@@ -172,9 +174,25 @@
...
+        public String path = getDefaultPath();
         public int maxSizeMb = 5;
         public int maxHistory = 1;
+
+        private static String getDefaultPath() {
+            if (!DiagnosticsHelper.isRpIntegration()) {
+                // this will be relative to the directory where agent jar is located
+                return DEFAULT_NAME;
+            }
+            if (DiagnosticsHelper.useAppSvcRpIntegrationLogging()) {
```
Comment (heyams) [inline agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/wasbootstrap/configuration/Configuration.java:188]: For sping cloud, which if statement does it fall through?

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/wasbootstrap/configuration/Configuration.java [hunk 2]
```diff
@@ -172,9 +174,25 @@
...
+                return StatusFile.getLogDir() + "/" + DEFAULT_NAME;
+            }
+            if (DiagnosticsHelper.useFunctionsRpIntegrationLogging() && !DiagnosticsHelper.isOsWindows()) {
+                return "/var/log/applicationinsights/" + DEFAULT_NAME;
```
Comment (heyams) [inline agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/wasbootstrap/configuration/Configuration.java:192]: 👍

File: agent/agent-bootstrap/src/test/java/com/microsoft/applicationinsights/agent/bootstrap/diagnostics/status/StatusFileTests.java [hunk 3]
```diff
@@ -193,7 +168,7 @@ public void putValueAndWriteOverwritesCurrentFile() throws Exception {
 
 
             final File tempFolder = this.tempFolder.newFolder();
-            StatusFile.directory = tempFolder.getAbsolutePath();
+            StatusFile.statusDir = tempFolder.getAbsolutePath();
```
Comment (heyams) [inline agent/agent-bootstrap/src/test/java/com/microsoft/applicationinsights/agent/bootstrap/diagnostics/status/StatusFileTests.java:171]: nit: statusFile.statusDir, 'status' seems redundant here.

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/wasbootstrap/configuration/ConfigurationBuilder.java [hunk 4]
```diff
@@ -173,6 +173,13 @@ private static Configuration loadConfigurationFile(Path agentJarPath) throws IOE
             }
         }
 
+        if (DiagnosticsHelper.isRpIntegration()) {
+            // users do not have write access to agent directory in rp integrations
```
Comment (heyams) [inline agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/wasbootstrap/configuration/ConfigurationBuilder.java:177]: are we going to make this comment public on azure docs?

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/wasbootstrap/configuration/Configuration.java (hunk 1 above)
Comment (trask) [reply agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/wasbootstrap/configuration/Configuration.java:188]: oh my, good point, fixed!

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/wasbootstrap/configuration/ConfigurationBuilder.java (hunk 4 above)
Comment (trask) [reply agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/wasbootstrap/configuration/ConfigurationBuilder.java:177]: i'm not sure about this specific comment, but we definitely need RP-specific azure docs on how to configure the agent

**Review by heyams — APPROVED** [review]

**Pull request merged**